boilerplate/openshift/custom-catalog-osd-operator/current-version-getter.sh

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+set -o xtrace
+
+# set -E
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+CERT_DIR="${REPO_ROOT}/service-account/rhcs-request/"
+CERT_FILE_NAME="custom-catalog-source-index-builder"
+PYXIS_ENDPOINT="https://pyxis.engineering.redhat.com"
+VERSIONS_DIR="${REPO_ROOT}/versions"
+
+
+function compare_versions () {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+function update_latest_version() {
+  # perform the API request
+  local channel_name=${1}
+  local version_in_repo=${2}
+  local version_file=${3}
+
+  local available_operator_versions=$(curl -X GET --key ${CERT_DIR}/${CERT_FILE_NAME}.key --cert ${CERT_DIR}/${CERT_FILE_NAME}.crt \
+    "${PYXIS_ENDPOINT}/v1/operators/bundles?page_size=100&organization=redhat-operators&filter=package==compliance-operator;in_index_img==true;channel_name==${channel_name}" | \
```

Comment:
compliance-operator needs to be parameterized here, yeah?

Reply:
yeah, good catch. I will parameterise it, fetch the name based on the name of the repo.